offlinevalidator/offlinevalidator.go

```diff
@@ -79,9 +86,19 @@ func printResult(valid *bool, logger *logger.Logger) {
 	}
 }
 
-func validateResources(resources map[string]template.Resource, specification *specification.Specification, sink *logger.Logger) (bool) {
+//func validateResources(resources map[string]template.Resource, specification *specification.Specification, sink *logger.Logger) bool {
+func validateResources(templ cloudformation.Template, origTempl template.Template, specification *specification.Specification, sink *logger.Logger) bool {
 	valid := true
-	for resourceName, resourceValue := range resources {
+
+	origResources := origTempl.Resources
+	newResources := templ.Resources
+
+	errDecode := mapstructure.Decode(newResources, &origResources)
+	if errDecode != nil {
+		//don't log
```

Comment:
When `//don't log` comment would be replaced by `context.Logger.Error(errDecode.Error())`, `context *context.Context` set as argument to function and `context` passed as it's value, the errDecode would log:

ERROR error(s) decoding:
...
* '[template.Resource name]' expected a map, got 'bool'
...

whenever a value of a property would be a boolean value (e.g. evaluated by !Equals intrinsic function). But after logging all the decoding errors, it would log if template is valid or not and eventually log the missing property as it should do and the error doesn't stand as obstacle of validation.

Reply:
Got it, I'd then provide that explanation for the comment.
BTW. Please use full and valid _English_ grammar in the comments (capital letters, dots etc.)

Reply:
Ok, I will fix it.